examples/shell-environments/flake.nix

```
@@ -0,0 +1,27 @@
+{
+  description = "Description for the project";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
+  };
+
+  outputs = inputs@{ flake-parts, ... }:
+    flake-parts.lib.mkFlake { inherit inputs; } {
+      systems = [ "x86_64-linux" "aarch64-darwin" "x86_64-darwin" ];
+      perSystem = { config, self', inputs', pkgs, system, ... }: {
+        devShells.default = pkgs.mkShell {
+          buildInputs = with pkgs; [ terraform wget bat git ];
```

```suggestion
          nativeBuildInputs = with pkgs; [ terraform wget bat git ];
```

This makes shell completions work.

Also, tl;dr needed for cross and strictDeps.

`buildInputs` is for what I would call "components" or things that can become part of the output. Inputs specified here won't be runnable when enabling `strictDeps` or cross compilation, because the cross compilation infrastructure will pick the variation that has the same build and host platform as the current derivation, which implies that these inputs have a `hostPlatform` that's not compatible with the `buildPlatform`, which is the platform we need for commands we execute.

Seems to complicated to explain here, but I typed it anyway so here you go.

---

Thanks, it's clera. I was not aware of it! I'll update it 💪🏻